omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_postgres_validator.rb

```diff
@@ -0,0 +1,322 @@
+class PostgresqlPreflightValidator < PreflightValidator
+  attr_reader :cs_pg_attr, :node_pg_attr
+  require "pg"
+
+  def initialize(node)
+    super
+
+    # Note that PrivateChef['postgresql'] will currently contain
+    # ONLY the settings specified in chef-server.rb plus some
+    # attributes set on initialization of the class.
+    @cs_pg_attr = PrivateChef['postgresql']
+
+    # Represents the recipe defaults
+    @node_pg_attr = node['private_chef']['postgresql']
+
+
+  end
+
+
+  def run!
+    verify_unchanged_external_flag
+
+    # Our additional validations only apply when a database server exists,
+    # which - for the time being - means we're only going to run these checks
+    # if external DB is configured.
+    return unless cs_pg_attr['external']
+    external_postgres_config_validation
+    # In the future, we should be able to run these
+    # on non-backend nodes:
+    common_validation
+    backend_validation
+  end
+
+
+  def external_postgres_config_validation
+    # For now we're only performing these checks for external pgsql mode:
+    return unless cs_pg_attr['external']
+    # vip must be set in chef-server.rb and it take reasonable precautions to
+    # ensure it wasn't set to a local host.
+    # Note that this will also prevent people from running external postgres on the same
+    # box as the chef backend. The idea isn't to block 100% of cases, but just to prevent a
+    # possible accidental misconfiguration.
+    #
+    if (cs_pg_attr['vip'].nil? ||
+        cs_pg_attr['vip'] == '::1' ||
+        cs_pg_attr['vip'] == '127.0.0.1'  ||
+        cs_pg_attr['vip'] == 'localhost')
+      fail_with err_missing_external_host
```

Comment:
Is it possible this check will get in our way later?  What if I just want to run postgresql myself but I still want it on the same machine?